src/components/Films/Films.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Select, Table, Col, Input, Typography } from 'antd';
+
+import { getFilmDataByGenre, getFilmDataByQuery } from '../Redux/Action';
+import { RootState } from '../../Reducers';
+import { columns } from '../Config/filmsTable.config';
```

Comment:
config should return a function which can have params like data and return table config, currenlty you just return columns.

For example

export const shipmentDetailsHistoryConfig = (
  items: IShipmentItem[],
  carrier: ICarrier,
  status: string,
) => {
  const columns = [
    {
      title: 'Date',
      dataIndex: 'createdAt',
      key: 'createdAt',
      render: (date: string) => {
        const newDate = moment(date).format('DD MMM YYYY h:mm A');
        return <span>{newDate.toLocaleString()}</span>;
      },
      sorter: (a: IShipmentItem, b: IShipmentItem) => a.createdAt.localeCompare(b.createdAt),
    },
    {
      title: 'Courier',
      dataIndex: 'carrierDetails',
      key: 'carrierDetails',
      render: (c: ICarrierDetails) => c.carrierName || '-',
      sorter: (a: ICarrierDetails, b: ICarrierDetails) =>
        a?.carrierName?.localeCompare(b.carrierName),
    },
    {
      title: 'Status',
      dataIndex: 'status',
      key: 'status',
      sorter: (a: any, b: any) => a.status.localeCompare(b.status),
    },
    {
      title: 'Courier Status',
      dataIndex: 'carrierStatus',
      key: 'carrierStatus',
      render: (c: ICarrier) => c.carrierStatus || '-',
      sorter: (a: ICarrier, b: ICarrier) => a.carrierStatus.localeCompare(b.carrierStatus),
    },
    {
      title: 'Details',
      dataIndex: 'status',
      key: 'details',
      sorter: (a: any, b: any) => a.status.localeCompare(b.status),
    },
  ];

  return {
    columns,
    rowKey: 'id',
    pagination: false,
    dataSource: [{ ...items, ...carrier, status }],
  };
};